Add queue.len() keyboard

lental opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

Adding queue.len() keyboards to QMK

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout (addition or update)
- [ ] Documentation

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

waffle87 commented on keyboards/dotlen/queue/keyboard.json:
```diff
@@ -0,0 +1,118 @@
+       "cols": ["D4", "D6", "D7", "B4", "B5", "B6", "C6", "C7", "F7", "F6", "F5", "F4", "F1", "F0", "E6"],
+      "rows": ["D5", "D3", "D2", "D1", "D0"]
+    },
+
+    "debounce": 5,
```
```suggestion
```

waffle87 commented on keyboards/dotlen/queue/keyboard.json:
```diff
@@ -0,0 +1,118 @@
+
+    "debounce": 5,
+    "encoder": {
+        "enabled": true,
+        "rotary": [
+            {"pin_a": "B0", "pin_b": "B7"}
+        ],
+        "resolution":5,
```
```suggestion
        "rotary": [
            {"pin_a": "B0", "pin_b": "B7"}
        ],
```

waffle87 commented on keyboards/dotlen/queue/keyboard.json:
```diff
@@ -0,0 +1,118 @@
+             {"label":"1", "matrix": [0, 1], "x":1, "y":0},
+             {"label":"2", "matrix": [0, 2], "x":2, "y":0},
+             {"label":"3", "matrix": [0, 3], "x":3, "y":0},
+             {"label":"4", "matrix": [0, 4], "x":4, "y":0,},
```
```suggestion
             {"label":"4", "matrix": [0, 4], "x":4, "y":0},
```

waffle87 commented on keyboards/dotlen/queue/keyboard.json:
```diff
@@ -0,0 +1,118 @@
+             {"label":"Ctrl", "matrix": [4, 8], "x":16, "y":4},
+             {"label":"left", "matrix": [4, 12], "x":17, "y":4},
+             {"label":"down", "matrix": [4, 13], "x":18, "y":4},
+             {"label":"right", "matrix": [4, 14], "x":19, "y":4},]
```
```suggestion
             {"label":"right", "matrix": [4, 14], "x":19, "y":4}]
```

waffle87 commented on keyboards/dotlen/queue/keyboard.json:
Please run `qmk format-json` on this file.

waffle87 commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,50 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
+#define _AL 2
+#define _LL 3
+
```
```suggestion
enum layers {
    _BL,
    _FL,
    _AL
};

```

waffle87 commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,50 @@
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS ),
+
+
+[_LL] = LAYOUT(
+  KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,          KC_TRNS, KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,
+  KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS ),
+
```
```suggestion
  KC_TRNS, KC_TRNS, KC_TRNS,            KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS )
```

waffle87 commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,50 @@
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_BL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_FL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_AL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_LL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)}
```
```suggestion
    [_AL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)}
```

waffle87 commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,50 @@
+  /* Keymap _BL: Base Layer (Default Layer)
+   */
+[_BL] = LAYOUT(
+  KC_GRAVE, KC_1   , KC_2   , KC_3, KC_4, KC_5   , KC_6  , KC_7   , KC_8   , KC_9 , KC_0, KC_MINUS,KC_EQUAL, KC_BSPC,KC_BSPC, KC_MUTE,
+  KC_TAB , KC_Q   , KC_W   , KC_E, KC_R   , KC_T  , KC_Y, KC_U   , KC_I   , KC_O   , KC_P   , KC_LEFT_BRACKET, KC_RIGHT_BRACKET, KC_BACKSLASH , KC_PGUP,
```
```suggestion
  KC_GRAVE, KC_1   , KC_2   , KC_3, KC_4, KC_5   , KC_6  , KC_7   , KC_8   , KC_9 , KC_0, KC_MINS,KC_EQL, KC_BSPC,KC_BSPC, KC_MUTE,
  KC_TAB , KC_Q   , KC_W   , KC_E, KC_R   , KC_T  , KC_Y, KC_U   , KC_I   , KC_O   , KC_P   , KC_LBRC, KC_RBRC, KC_BSLS , KC_PGUP,
```

waffle87 commented on keyboards/dotlen/queue/readme.md:
```diff
@@ -0,0 +1,28 @@
+
+![Queue.len()](https://i.imgur.com/8JKbm6Ph.png)
+
+my 65% keyboard with multiple layout support
```
Missing layout support from the firmware?

lental replied:
good call. The PCB has layout support, but I didn't write out all  of the LAYOUTs.  I renamed the layout in keyboard.json to be LAYOUT_all as  suggested in PR Checklist. Can I come  back with additional layouts, or just take out the comment altogether?

lental replied:
never mind - I added two of the supported layouts for the keyboard.  Let me know if you were looking for something different

waffle87 reviewed (changes requested):
Please run `qmk format-json` on this file.

drashna commented on keyboards/dotlen/queue/keyboard.json:
```diff
@@ -0,0 +1,263 @@
+            ]
+        }
+    }
+}
```
```suggestion
}

```

zvecr replied:
```suggestion
    "layout_aliases": {
        "LAYOUT_all": "LAYOUT_triple_space"
    },
    "layouts": {
        "LAYOUT_standard_space": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "1", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "2", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "3", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "4", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "5", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "6", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "7", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "8", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "9", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": "0", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "-", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "backspace", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "split", "matrix": [0, 14], "x": 14, "y": 0},
                {"label": "rotary", "matrix": [2, 14], "x": 15, "y": 0, "encoder":0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "Bksp", "matrix": [1, 13], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "pgup", "matrix": [1, 14], "x": 15, "y": 1},
                {"label": "Caps Lock", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "A", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 12], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "pgup", "matrix": [2, 13], "x": 15, "y": 2},
                {"label": "split", "matrix": [3, 0], "x": 0, "y": 3, "w": 1.25},
                {"label": "Shift", "matrix": [3, 1], "x": 1.25, "y": 3, "w": 1},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Up", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "pgup", "matrix": [3, 14], "x": 15, "y": 3},
                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.25},
                {"label": "Win", "matrix": [4, 1], "x": 1.25, "y": 4, "w": 1.25},
                {"label": "Alt", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.25},
                {"matrix": [4, 3], "x": 3.75, "y": 4, "w": 6.25},
                {"label": "Alt", "matrix": [4, 6], "x": 10, "y": 4},
                {"label": "Win", "matrix": [4, 7], "x": 11, "y": 4},
                {"label": "Ctrl", "matrix": [4, 8], "x": 12, "y": 4},
                {"label": "left", "matrix": [4, 12], "x": 13, "y": 4},
                {"label": "down", "matrix": [4, 13], "x": 14, "y": 4},
                {"label": "right", "matrix": [4, 14], "x": 15, "y": 4}
            ]
        },
        "LAYOUT_triple_space": {
            "layout": [
                {"label": "Esc", "matrix": [0, 0], "x": 0, "y": 0},
                {"label": "1", "matrix": [0, 1], "x": 1, "y": 0},
                {"label": "2", "matrix": [0, 2], "x": 2, "y": 0},
                {"label": "3", "matrix": [0, 3], "x": 3, "y": 0},
                {"label": "4", "matrix": [0, 4], "x": 4, "y": 0},
                {"label": "5", "matrix": [0, 5], "x": 5, "y": 0},
                {"label": "6", "matrix": [0, 6], "x": 6, "y": 0},
                {"label": "7", "matrix": [0, 7], "x": 7, "y": 0},
                {"label": "8", "matrix": [0, 8], "x": 8, "y": 0},
                {"label": "9", "matrix": [0, 9], "x": 9, "y": 0},
                {"label": "0", "matrix": [0, 10], "x": 10, "y": 0},
                {"label": "-", "matrix": [0, 11], "x": 11, "y": 0},
                {"label": "+", "matrix": [0, 12], "x": 12, "y": 0},
                {"label": "backspace", "matrix": [0, 13], "x": 13, "y": 0},
                {"label": "split", "matrix": [0, 14], "x": 14, "y": 0},
                {"label": "rotary", "matrix": [2, 14], "x": 15, "y": 0, "encoder":0},
                {"label": "Tab", "matrix": [1, 0], "x": 0, "y": 1, "w": 1.5},
                {"label": "Q", "matrix": [1, 1], "x": 1.5, "y": 1},
                {"label": "W", "matrix": [1, 2], "x": 2.5, "y": 1},
                {"label": "E", "matrix": [1, 3], "x": 3.5, "y": 1},
                {"label": "R", "matrix": [1, 4], "x": 4.5, "y": 1},
                {"label": "T", "matrix": [1, 5], "x": 5.5, "y": 1},
                {"label": "Y", "matrix": [1, 6], "x": 6.5, "y": 1},
                {"label": "U", "matrix": [1, 7], "x": 7.5, "y": 1},
                {"label": "I", "matrix": [1, 8], "x": 8.5, "y": 1},
                {"label": "O", "matrix": [1, 9], "x": 9.5, "y": 1},
                {"label": "P", "matrix": [1, 10], "x": 10.5, "y": 1},
                {"label": "{", "matrix": [1, 11], "x": 11.5, "y": 1},
                {"label": "}", "matrix": [1, 12], "x": 12.5, "y": 1},
                {"label": "Bksp", "matrix": [1, 13], "x": 13.5, "y": 1, "w": 1.5},
                {"label": "pgup", "matrix": [1, 14], "x": 15, "y": 1},
                {"label": "Caps Lock", "matrix": [2, 0], "x": 0, "y": 2, "w": 1.75},
                {"label": "A", "matrix": [2, 1], "x": 1.75, "y": 2},
                {"label": "S", "matrix": [2, 2], "x": 2.75, "y": 2},
                {"label": "D", "matrix": [2, 3], "x": 3.75, "y": 2},
                {"label": "F", "matrix": [2, 4], "x": 4.75, "y": 2},
                {"label": "G", "matrix": [2, 5], "x": 5.75, "y": 2},
                {"label": "H", "matrix": [2, 6], "x": 6.75, "y": 2},
                {"label": "J", "matrix": [2, 7], "x": 7.75, "y": 2},
                {"label": "K", "matrix": [2, 8], "x": 8.75, "y": 2},
                {"label": "L", "matrix": [2, 9], "x": 9.75, "y": 2},
                {"label": ":", "matrix": [2, 10], "x": 10.75, "y": 2},
                {"label": "\"", "matrix": [2, 11], "x": 11.75, "y": 2},
                {"label": "Enter", "matrix": [2, 12], "x": 12.75, "y": 2, "w": 2.25},
                {"label": "pgup", "matrix": [2, 13], "x": 15, "y": 2},
                {"label": "split", "matrix": [3, 0], "x": 0, "y": 3, "w": 1.25},
                {"label": "Shift", "matrix": [3, 1], "x": 1.25, "y": 3, "w": 1},
                {"label": "Z", "matrix": [3, 2], "x": 2.25, "y": 3},
                {"label": "X", "matrix": [3, 3], "x": 3.25, "y": 3},
                {"label": "C", "matrix": [3, 4], "x": 4.25, "y": 3},
                {"label": "V", "matrix": [3, 5], "x": 5.25, "y": 3},
                {"label": "B", "matrix": [3, 6], "x": 6.25, "y": 3},
                {"label": "N", "matrix": [3, 7], "x": 7.25, "y": 3},
                {"label": "M", "matrix": [3, 8], "x": 8.25, "y": 3},
                {"label": "<", "matrix": [3, 9], "x": 9.25, "y": 3},
                {"label": ">", "matrix": [3, 10], "x": 10.25, "y": 3},
                {"label": "?", "matrix": [3, 11], "x": 11.25, "y": 3},
                {"label": "Shift", "matrix": [3, 12], "x": 12.25, "y": 3, "w": 1.75},
                {"label": "Up", "matrix": [3, 13], "x": 14, "y": 3},
                {"label": "pgup", "matrix": [3, 14], "x": 15, "y": 3},
                {"label": "Ctrl", "matrix": [4, 0], "x": 0, "y": 4, "w": 1.25},
                {"label": "Win", "matrix": [4, 1], "x": 1.25, "y": 4, "w": 1.25},
                {"label": "Alt", "matrix": [4, 2], "x": 2.5, "y": 4, "w": 1.25},
                {"matrix": [4, 3], "x": 3.75, "y": 4, "w": 2.25},
                {"matrix": [4, 4], "x": 6, "y": 4, "w": 1.25},
                {"matrix": [4, 5], "x": 7.25, "y": 4, "w": 2.75},
                {"label": "Alt", "matrix": [4, 6], "x": 10, "y": 4},
                {"label": "Win", "matrix": [4, 7], "x": 11, "y": 4},
                {"label": "Ctrl", "matrix": [4, 8], "x": 12, "y": 4},
                {"label": "left", "matrix": [4, 12], "x": 13, "y": 4},
                {"label": "down", "matrix": [4, 13], "x": 14, "y": 4},
                {"label": "right", "matrix": [4, 14], "x": 15, "y": 4}
            ]
        }
    }
}
```

drashna commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,42 @@
+    [_FL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_AL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)}
+};
+#endif
```
```suggestion
#endif

```

drashna commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,42 @@
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_BL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_FL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)},
+    [_AL] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU)}
```
Note that you can use transparency here: 
```suggestion
    [_FL] = { ENCODER_CCW_CW(_______, _______)},
    [_AL] = { ENCODER_CCW_CW(_______, _______)}
```

lental commented:
Hey! will try to keep this open.

The request is still valid - I think I addressed all feeback.. Please let me know if there's anything I should fix

Best,
Michael

lental commented:
Hey bumping - is there anything I'm missing? I think I addressed all comments. Let me know if there's something I'm supposed to close out.

Thanks,
Michael

drashna commented:
> is there anything I'm missing?

Just a second review.  Generally, we want/require 2 reviews before a PR gets merged. It can be easy to overlook things, and different areas of expertise, etc, so a second pair of eyes helps to avoid issues down the line.

zvecr commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,42 @@
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  /* Keymap _BL: Base Layer (Default Layer)
+   */
```
```suggestion
```

zvecr commented on keyboards/dotlen/queue/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: GPL-2.0-or-later */
```
```suggestion
// Copyright 2025 lental
// SPDX-License-Identifier: GPL-2.0-or-later
```

zvecr commented on keyboards/dotlen/queue/readme.md:
```diff
@@ -0,0 +1,28 @@
+
+![Queue.len()](https://i.imgur.com/8JKbm6Ph.png)
+
+my 65% keyboard with multiple layout support
```
```suggestion
A 65% keyboard with multiple layout support.
```

zvecr reviewed (changes requested):
This should fix the layout presentation, as shown in the lint output.
